frontend/src/pages/modelServing/screens/projects/kServeModal/ManageKServeModal.tsx

```diff
@@ -59,29 +83,43 @@ const ManageKServeModal: React.FC<ManageKServeModalProps> = ({
   const [createDataServingRuntime, setCreateDataServingRuntime, resetDataServingRuntime, sizes] =
     useCreateServingRuntimeObject(editInfo?.servingRuntimeEditInfo);
   const [createDataInferenceService, setCreateDataInferenceService, resetDataInferenceService] =
-    useCreateInferenceServiceObject(editInfo?.inferenceServiceEditInfo);
+    useCreateInferenceServiceObject(
+      editInfo?.inferenceServiceEditInfo,
+      editInfo?.servingRuntimeEditInfo?.servingRuntime,
+      editInfo?.secrets,
+    );
   const [acceleratorProfileState, setAcceleratorProfileState, resetAcceleratorProfileData] =
     useServingAcceleratorProfile(
       editInfo?.servingRuntimeEditInfo?.servingRuntime,
       editInfo?.inferenceServiceEditInfo,
     );
 
+  const namespace =
+    projectContext?.currentProject.metadata.name || createDataInferenceService.project;
+
+  const [allowCreate] = useAccessReview({
+    ...accessReviewResource,
+    namespace,
+  });
+
   const [actionInProgress, setActionInProgress] = React.useState(false);
   const [error, setError] = React.useState<Error | undefined>();
+  const [alertVisible, setAlertVisible] = React.useState(true);
   const isInferenceServiceNameWithinLimit =
     translateDisplayNameForK8s(createDataInferenceService.name).length <= 253;
 
+  const isAuthorinoEnabled = useIsAreaAvailable(SupportedArea.K_SERVE_AUTH).status;
+
+  const currentProjectName = projectContext?.currentProject.metadata.name;
```

Comment:
Do this earlier in the function so that you can use `currentProjectName` here:
```
const namespace =
    projectContext?.currentProject.metadata.name || createDataInferenceService.project;
```